packages/graphiql/src/components/DocExplorer.js

```diff
@@ -142,6 +168,24 @@ export class DocExplorer extends React.Component {
 
   // Public API
   showDoc(typeOrField) {
+    if (!typeOrField) { return; }
+
+    if (this.props.onShowDoc) {
+      let typeName;
+      let fieldName;
+      if (isType(typeOrField)) {
+        typeName = typeOrField.name;
+        fieldName = '';
+      } else {
+        fieldName = typeOrField.name;
+        const navStack = this.state.navStack;
+        const lastState = navStack[navStack.length - 1];
+        typeName = lastState.name;
+      }
+
+      this.props.onShowDoc(typeName, fieldName)
```

Comment:
On second thoughts, I think `this.props.onShowDoc` should pass `typeOrField` directly to user code; it should be up to the user code to decompress it into the relevant typeName/fieldName/directiveName/etc. You can move the code above this to the example to show how to do it, perhaps.

Reply:
Sure, that makes sense. I'm not sure how to include more dependencies in the example, since it just uses `<script>` tags - do you know how I could import the `graphql` package so that I can parse the `typeOrField` data?